curp/src/server/curp_node.rs

```diff
@@ -350,13 +359,13 @@ impl<C: 'static + Command> CurpNode<C> {
             // a sync is either triggered by an heartbeat timeout event or when new log entries arrive
             tokio::select! {
                 _now = ticker.tick() => {
-                    if hb_opt {
-                        hb_opt = false;
-                        continue;
+                    hb_opt = false;
+                }
+                res = tokio::time::timeout(batch_timeout, listener) => {
+                    if let Err(_e) = res {
+                        hb_opt = true;
                     }
```

Comment:
If the `batch_timeout` expires while there is not any log generated during this period, it will send an empty append_entries request to followers. That's not what we want. Whether or not a request is sent depends on two factors: (hb_opt, ae.entries.is_empty).